pkg/cmd/target/set.go

```diff
@@ -169,3 +192,114 @@ var TargetSetCmd = &cobra.Command{
 		return nil
 	},
 }
+
+func handleTargetJSON(data []byte) error {
+	ctx := context.Background()
+
+	apiClient, err := apiclient_util.GetApiClient(nil)
+	if err != nil {
+		return err
+	}
+	var selectedTarget *target_view.TargetView
+	err = parseJSON(data, &selectedTarget)
+	if err != nil {
+		return fmt.Errorf("failed to parse input: %w", err)
+	}
+
+	if selectedTarget.Name == "" {
+		return errors.New("invalid input: 'name' field is required")
+	}
+	if selectedTarget.Options == "" {
+		return errors.New("option fields are required to setup your target")
+	}
+	targetManifest, res, err := apiClient.ProviderAPI.GetTargetManifest(ctx, selectedTarget.ProviderInfo.Name).Execute()
+	if err != nil {
+		return apiclient_util.HandleErrorResponse(res, err)
+	}
```

Comment:
@Tpuljak i am still getting the error 
`FATA[0002] provider not found: provider not found`, it seems i have to find a way to extract the provider name from the `selectedTarget` variable .

Reply:
@Philip-21 the provider name should be part of the target config
![Screenshot 2024-12-10 at 14 11 42](https://github.com/user-attachments/assets/1bc0fd93-896d-426d-ab1c-3ee7844641ee)

Reply:
is there  anything else is should adjust, or the codes are  fine ?

Reply:
Push everything so I can do a final review

Reply:
i have pushed already